cluster/src/cluster_impl.rs

```diff
@@ -134,22 +140,81 @@ impl EventHandler for Inner {
-                            .await?;
+                    for table_info in &shard_tables.tables {
+                        let req = OpenTableRequest {
+                            catalog_name: DEFAULT_CATALOG.to_string(),
+                            schema_name: table_info.schema_name.to_string(),
+                            schema_id: table_info.schema_id.into(),
+                            table_name: table_info.name.to_string(),
+                            table_id: table_info.id.into(),
+                            // TODO: is this hard code appropriate?
+                            engine: ANALYTIC_ENGINE_TYPE.to_string(),
+                        };
+                        let table = self.table_engine.open_table(req).await?;
+                        if let Some(table) = table {
+                            self.table_manager
+                                .add_shard_table(table_info.to_owned(), table)?;
+                        }
                     }
                 }
                 self.table_manager.update_table_info(&resp.shard_tables);
 
                 Ok(())
             }
-            ActionCmd::CreateTableCmd(_) => todo!(),
-            ActionCmd::DropTableCmd(_)
-            | ActionCmd::MetaNoneCmd(_)
-            | ActionCmd::MetaCloseCmd(_)
-            | ActionCmd::MetaSplitCmd(_)
-            | ActionCmd::MetaChangeRoleCmd(_) => {
+            ActionCmd::CreateTableCmd(_create_cmd) => {
+                // TODO: Modify CreateTableCmd to alias with CreateTableRequest.
+                todo!()
+            }
+            ActionCmd::DropTableCmd(drop_cmd) => {
+                let schema_id = self
+                    .table_manager
+                    .get_schema_id(DEFAULT_CATALOG, &drop_cmd.schema_name)
+                    .context(SchemaNotFound {
+                        schema_name: drop_cmd.schema_name.to_owned(),
+                    })?;
+                let req = DropTableRequest {
+                    catalog_name: DEFAULT_CATALOG.to_owned(),
+                    schema_name: drop_cmd.schema_name.to_owned(),
+                    schema_id: schema_id.into(),
+                    table_name: drop_cmd.name.to_string(),
+                    // TODO: is this hard code appropriate?
+                    engine: ANALYTIC_ENGINE_TYPE.to_string(),
+                };
+
+                self.table_engine.drop_table(req).await?;
+
+                Ok(())
+            }
+            ActionCmd::MetaNoneCmd(_) => Ok(()),
+            ActionCmd::MetaCloseCmd(close_cmd) => {
+                let mut execute_errors = vec![];
+
+                for shard in &close_cmd.shard_ids {
+                    let tokens = self.table_manager.tokens_by_shard(*shard);
+                    for token in tokens {
+                        let req = CloseTableRequest {
+                            catalog_name: DEFAULT_CATALOG.to_string(),
+                            schema_name: "TODO".to_string(),
```

Comment:
Maybe we should use empty string and add TODO comment : Remove `schema_name` and `table_name` from `CloseTableRequest` in the future.